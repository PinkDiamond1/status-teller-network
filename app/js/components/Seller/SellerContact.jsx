import React, {Component, Fragment} from 'react';
import {FormGroup, Label} from 'reactstrap';
import Form from 'react-validation/build/form';
import Input from 'react-validation/build/input';
import PropTypes from 'prop-types';
import {required} from "../../validators";

class SellerContact extends Component {
  changeNickname(e) {
    this.props.changeNickname(e.target.value);
  }
  changeContactCode(e) {
    this.props.changeContactCode(e.target.value);
  }

  render() {
    return (
      <Fragment>
        <h2>Your name and how to contact you</h2>
        <p>What would be the best way for the buyer to contact you</p>

        <Form>
          <FormGroup>
            <Label for="nickname">Nickname</Label>
            <Input type="text" name="nickname" id="nickname" value={this.props.nickname} className="form-control"
                   onChange={(e) => this.changeNickname(e)} validations={[required]}/>
          </FormGroup>
          <FormGroup>
            <Label for="contactCode">Status contact code or Status ENS name</Label>
            <Input type="text" name="contactCode" id="contactCode" value={this.props.contactCode}
                   className="form-control" onChange={(e) => this.changeContactCode(e)}  validations={[required]}/>
          </FormGroup>
        {(!this.props.nickname || !this.props.contactCode) && <p className="text-info">Enter a location to move to the next page</p>}
        </Form>
      </Fragment>
    );
  }
}

SellerContact.propTypes = {
  changeNickname: PropTypes.func,
  changeContactCode: PropTypes.func,
  nickname: PropTypes.string,
  contactCode: PropTypes.string
};


export default SellerContact;
