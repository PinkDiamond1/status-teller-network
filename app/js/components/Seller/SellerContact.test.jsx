import React from "react";
import { shallow } from "enzyme";

import SellerContact from "./SellerContact";

describe('SellerContact', () => {
  it('should render correctly', () => {
    const component = shallow(<SellerContact changeNickname={() => undefined} changeContactCode={() => undefined}
                                             nickname={"nickname"} contractCode={"contractCode"} />);
  
    expect(component).toMatchSnapshot();
  });
});