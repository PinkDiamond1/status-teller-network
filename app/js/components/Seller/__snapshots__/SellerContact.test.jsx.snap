// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SellerContact should render correctly 1`] = `
<Fragment>
  <h2>
    Your name and how to contact you
  </h2>
  <p>
    What would be the best way for the buyer to contact you
  </p>
  <Form(t)>
    <FormGroup
      tag="div"
    >
      <Label
        for="nickname"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Nickname
      </Label>
      <Control(s)
        className="form-control"
        id="nickname"
        name="nickname"
        onChange={[Function]}
        type="text"
        validations={
          Array [
            [Function],
          ]
        }
        value="nickname"
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Label
        for="contactCode"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Status contact code or Status ENS name
      </Label>
      <Control(s)
        className="form-control"
        id="contactCode"
        name="contactCode"
        onChange={[Function]}
        type="text"
        validations={
          Array [
            [Function],
          ]
        }
      />
    </FormGroup>
    <p
      className="text-info"
    >
      Enter a location to move to the next page
    </p>
  </Form(t)>
</Fragment>
`;
